content/en/docs/concepts/extend-kubernetes/_index.md

```diff
@@ -83,127 +81,163 @@ Automation generally works with any Kubernetes cluster, including hosted
-![Flowchart for Extension](/docs/concepts/extend-kubernetes/flowchart.png)
+<!-- image source for flowchart: https://docs.google.com/drawings/d/1sdviU6lDz4BpnzJNHfNpQrqI9F19QZ07KnhnxVrp2yg/edit -->
+
+{{< figure src="/docs/concepts/extend-kubernetes/flowchart.png"
+    alt="Flowchart with questions about use cases and guidance for implementers. Green circles indicate yes; red circles indicate no."
+    class="diagram-large" caption="Flowchart guide to select an extension approach" >}}
+
+---
 
-## API Extensions
+## Client extensions
 
-### User-Defined Types
+Plugins for kubectl are separate binaries that add or replace the behavior of specific subcommands.
+The `kubectl` tool can also integrate with [credential plugins](/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins)
+These extensions only affect a individual user's local environment, and so cannot enforce site-wide policies.
 
-Consider adding a Custom Resource to Kubernetes if you want to define new controllers, application
+If you want to extend the `kubectl` tool, read [Extend kubectl with plugins](/docs/tasks/extend-kubectl/kubectl-plugins/).
+
+## API extensions
+
+### Custom resource definitions
+
+Consider adding a _Custom Resource_ to Kubernetes if you want to define new controllers, application
 configuration objects or other declarative APIs, and to manage them using Kubernetes tools, such
 as `kubectl`.
 
-Do not use a Custom Resource as data storage for application, user, or monitoring data.
-
 For more about Custom Resources, see the
-[Custom Resources concept guide](/docs/concepts/extend-kubernetes/api-extension/custom-resources/).
+[Custom Resources](/docs/concepts/extend-kubernetes/api-extension/custom-resources/) concept guide.
+
+### API aggregation layer
 
+You can use Kubernetes' [API Aggregation Layer](/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/)
+to integrate the Kubernetes API with additional services such as for [metrics](/docs/tasks/debug/debug-cluster/resource-metrics-pipeline/).
 
-### Combining New APIs with Automation
+### Combining new APIs with automation
 
-The combination of a custom resource API and a control loop is called the
-[Operator pattern](/docs/concepts/extend-kubernetes/operator/). The Operator pattern is used to manage
-specific, usually stateful, applications. These custom APIs and control loops can also be used to
-control other resources, such as storage or policies.
+A combination of a custom resource API and a control loop is called the
+{{< glossary_tooltip term_id="controller" text="controllers" >}} pattern. If your controller takes
+the place of a human operator deploying infrastructure based on a desired state, then the controller
+may also be following the {{< glossary_tooltip text="operator pattern" term_id="operator-pattern" >}}.
+The Operator pattern is used to manage specific applications; usually, these are applications that
```

Comment:
nit: could remove the semi-colon and shorten the sentence

The Operator pattern is used to manage applications which maintain state and require care in how they are managed.